src/python/pants/engine/internals/graph.py

```diff
@@ -366,8 +366,9 @@ def _target_parametrizations(
     target_type: type[Target],
     union_membership: UnionMembership,
 ) -> _TargetParametrization:
-    first, *rest = Parametrize.expand(address, target_adaptor.kwargs)
-    if rest:
+    expanded_parametrizations = tuple(Parametrize.expand(address, target_adaptor.kwargs))
+    first_address, first_kwargs = expanded_parametrizations[0]
+    if first_address != address:
```

Comment:
This will do an extra address comparison.

Maybe adjust `Parametrize.expand` to return an empty iterator if the target wasn't parametrized?

Reply:
That causes issues else were, where we rely on the fact that we get back also unparametrized result to use inline.

However, as the address returned in case of unparametrized targets is the same object passed in, we can use the `is` operator instead for a cheaper check.